tests/console/validate_autoyast.pm

```diff
@@ -0,0 +1,56 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: smoke test for autoyast post-installation
+# Maintainer: Yiannis Bonatakis <ybonatakis@suse.com>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use utils;
+use y2logsstep;
+use version_utils 'is_opensuse';
+
+sub run {
+    my $self = shift;
+    $self->select_serial_terminal;
+
+    record_info('INFO', 'Check environment');
+    assert_script_run "env | grep \"SHELL=/bin/bash\"";
+    assert_script_run "env | grep \"HOME=\/root\"";
+    assert_script_run "env | grep \"OSTYPE=linux\"";
+
+    record_info('INFO', 'Check user');
+    assert_script_run "id -u $username |grep 1000";
+
+    if (not is_opensuse()) {
+        # opensuse_gnome has simpler configuration and timezone or network are not defined
+        record_info('INFO', 'Check firewall is not enabled and not running');
+        my $service = opensusebasetest::firewall();
+        assert_script_run qq{systemctl status  ($service) | grep \"inactive \(dead\)\"};
+        assert_script_run qq{systemctl list-unit-files | grep disabled | grep firewalld};
+
+        record_info('INFO', 'Verify networking');
+        assert_script_run "ip link show | grep -E \"(ens|eth)[0-9]\" | grep UP";
+    }
+    else {
+        record_info('INFO', 'Check firewall is enabled and running');
+        assert_script_run qq{systemctl status firewalld.service | grep \"active \(running\)\"};
```

Comment:
Ok, small remark. In SUT we have pipe configured to fail if any command in the pipe fails. It means that if ` systemctl status firewalld.service` returns non-zero code, whole command will fail. In this case grep becomes useless, as `systemctl status` returns 0 only in case service is running.

Reply:
also check latest PR